### ISSUE TITLE
Feature request: automatically interlink, without needing bracket notation

### ISSUE THREAD

**Issue by JonathanReeve** [seongjaelee/notational-velocity#47]
It'd be nice not to have to surround interlinks with brackets. If nvatom can index all notes, it'd be great if it could automatically link those note titles, without having to have the user identify links. (Like in Tomboy Notes or Vim-Notes)

**Comment by seongjaelee** [comment]
Thanks for your suggestion. 

This is doable, but I am against this feature.

If we have a document whose title is very common, something like "diary", "note", or "google", then all related words will be converted into a link.

nvALT also does not have feature. Wiki markup syntaxes I know also does not have this feature.

**Comment by ghost** [comment]
Good idea. It helps the user by eliminating the need to manually interlink documents. (while retaining the feature if you want to manually link documents.) This allows links to exist as meta-data apart from the actual text of the documents. This allows users the ability to interlink documents without using link demarcations such as `[]` or `[[]]`.

> If we have a document whose title is very common, something like "diary", "note", or "google", then all related words will be converted into a link.

Why is this a problem? Don't most people create more descriptive note titles? So the likelihood of common words being used as titles is rather low. (I would imagine)

By automatically underlining and/or coloring a word to indicate a link, the user is saved the trouble of manually creating links. And to create a new file, simply make a selection of word or group of words, and run the follow link command to create a new file. Once this new file is saved, the previous file will now have a new link automatically created for the user.

**Comment by seongjaelee** [comment]
I thought more about this feature today, but I am still reluctant to it - I'm very conservative, and I'll prioritize the features already in notational velocity or nvalt.

Some thoughts:
1. If we have "watch.md", "watch tower.md", and "crime watch.md". Let's say the text of "test.md" file is "crime watch tower. Let's play Overwatch. Watch out. Apple watches are expensive." What should be highlighted? What if the user wants to highlight "crime [[watch tower]]. Let's play Overwatch. [[Watch]] out. Apple [[watch]]es are expensive."?
2. To implement it, I need to scan the body and try to find all matching texts based on existing note pathes. We can create two subclasses of Interlink.coffee, and support an option to select one.
3. Actually, such interlink feature can be written as a separate plugin. The only drawback I can think of is that the user have to specify the base directory twice (in notational-velocity package options and interlink package options). (Actually, this can be applied to the current interlink...)
